sql/parser2/eval.go

```diff
@@ -20,31 +20,308 @@ package parser2
 import (
 	"fmt"
 	"math"
+	"reflect"
 	"strconv"
-
-	"github.com/cockroachdb/cockroach/sql/sqlwire"
 )
 
-type opType int
+// TODO(pmattis):
+//
+// - Support tuples (i.e. []Datum) and tuple operations (a IN (b, c, d)).
+//
+// - Support decimal arithmetic.
+
+// A Datum holds either a bool, int64, float64, string or []Datum.
+type Datum interface {
+	IsNull() bool
+	ToBool() (Datum, error)
```

Comment:
looks like all calls to this are followed by a type assertion to `dbool`. can you make this just return `dbool`?